lib/rules/unicode-bom.js

```diff
@@ -43,9 +43,14 @@ module.exports = {
             Program: function checkUnicodeBOM(node) {
 
                 const sourceCode = context.getSourceCode(),
-                    location = { column: 0, line: 1 },
+                    firstToken = sourceCode.getFirstToken(node),
+                    location = { start: { column: 0, line: 1 } },
                     requireBOM = context.options[0] || "never";
 
+                if (firstToken) {
+                    location.end = firstToken.loc.end;
+                }
```

Comment:
I'm not sure if it makes sense to highlight a potentially large range because of the missing/required BOM character.

For example, this is how it would look like on the rule source file itself:

![image](https://user-images.githubusercontent.com/44349756/126225456-81a75686-e6a3-4154-99f8-f0accea4966c.png)

Thoughts about highlighting only the first character, as proposed in https://github.com/eslint/eslint/issues/12334#issuecomment-655085810?

Reply:
Also, this is basically the same situation (reporting the beginning of the file) as [one case](https://github.com/eslint/eslint/blob/6641d88e17d952a8e51df5e0d3882a842d4c3f35/tests/lib/rules/consistent-return.js#L312) in the `consistent-return` rule where we've decided to keep reporting only the start loc?

Reply:
I think it's better with keeping only the start loc (same as `consistent-return`)

Reply:
I updated the test cases to assert on reporting location.